### PR TITLE
replicator fd leak: close the server stream on failure cases so that the tcp connection can be actually closed

### PR DESCRIPTION
**Pull request by datoug**
as title. make sure the server calls close on failure cases.

### CODE REVIEW

File: services/replicator/replicator.go [hunk 1]
```diff
@@ -219,6 +219,10 @@ func (r *Replicator) OpenReplicationReadStreamHandler(w http.ResponseWriter, req
 		}).Error("Can't open store host read stream")
 		r.m3Client.IncCounter(metrics.OpenReplicationReadScope, metrics.ReplicatorCreateOutStreamFailure)
 		r.m3Client.IncCounter(metrics.OpenReplicationReadScope, metrics.ReplicatorFailures)
+
+		// Must close the connection on server side because closing on client side doesn't actually close the
+		// underlying TCP connection
+		inStream.Done()
```
Comment (kirg) [inline services/replicator/replicator.go]: Probably better do a "defer inStream.Done()" immediately after the open and if-error block.

File: services/replicator/replicator.go [hunk 2]
```diff
@@ -276,6 +280,10 @@ func (r *Replicator) OpenReplicationRemoteReadStreamHandler(w http.ResponseWrite
 		}).Error("Can't open remote replication read stream")
 		r.m3Client.IncCounter(metrics.OpenReplicationRemoteReadScope, metrics.ReplicatorCreateOutStreamFailure)
 		r.m3Client.IncCounter(metrics.OpenReplicationRemoteReadScope, metrics.ReplicatorFailures)
+
+		// Must close the connection on server side because closing on client side doesn't actually close the
+		// underlying TCP connection
+		inStream.Done()
```
Comment (kirg) [inline services/replicator/replicator.go]: defer inStream.Done() .. like above.

File: services/replicator/replicator.go (hunk 1 above)
Comment (datoug) [inline services/replicator/replicator.go]: problem is in regular cases stream is closed in the inconnection goroutine. If we do a defer, two goroutines might try to close the stream. From the websocket API the close functions are not thread-safe, also I'm not sure about the behavior of double close.

Comment (kirg) [inline services/replicator/replicator.go]: yes, close isn't thread-safe. i see .. i hadn't noticed that; this is fine then.

**Pull request merged**